ray-curator/tests/stages/test_file_partitioning.py

```diff
@@ -132,32 +147,31 @@ def test_process_with_limit(self, empty_task: _EmptyTask):
 
         # Should only return 3 file groups due to limit
         assert len(result) == 3
-        assert result[0].data == ["/path/file0.jsonl", "/path/file1.jsonl"]
-        assert result[1].data == ["/path/file2.jsonl", "/path/file3.jsonl"]
-        assert result[2].data == ["/path/file4.jsonl", "/path/file5.jsonl"]
+        assert result[0].data == test_files[:2]
+        assert result[1].data == test_files[2:4]
+        assert result[2].data == test_files[4:6]
 
         # Verify metadata
         for i, task in enumerate(result):
             assert task.task_id == f"file_group_{i}"
             assert task._metadata["partition_index"] == i
             assert task._metadata["total_partitions"] == 5  # Total partitions before limit
 
-    def test_process_with_limit_single_partition(self, empty_task: _EmptyTask):
+    def test_process_with_limit_single_partition(self, empty_task: _EmptyTask, tmp_path: Path):
         """Test limit when all files would be in a single partition."""
-        test_files = [f"/path/file{i}.jsonl" for i in range(5)]
+        test_files = _create_test_jsonl_files(tmp_path, num_files=5, subdir="path")
         stage = FilePartitioningStage(
             file_paths=test_files,
-            limit=1,  # Limit to 1 group, and all files would be in one group anyway
+            limit=1,  # Limit to 1 group, TODO: Ask ayush why this is the behavior
```

Comment:
Can we resolve any tagged comments/open follow-up issues for open TODO items?

Reply:
Added here: https://github.com/NVIDIA-NeMo/Curator/issues/949  and https://github.com/NVIDIA-NeMo/Curator/issues/948  , also tagged 948 in comments